akka-remote/src/main/scala/akka/remote/serialization/ArteryControlMessageSerializer.scala

```diff
@@ -0,0 +1,200 @@
+    case ActorRefCompressionAdvertisementAck(from, id)      ⇒ serializeCompressionTableAdvertisementAck(from, id)
+    case adv: ClassManifestCompressionAdvertisement         ⇒ serializeCompressionAdvertisement(adv)(identity)
+    case ClassManifestCompressionAdvertisementAck(from, id) ⇒ serializeCompressionTableAdvertisementAck(from, id)
+    case env: SystemMessageDelivery.SystemMessageEnvelope   ⇒ serializeSystemMessageEnvelope(env)
+    case SystemMessageDelivery.Ack(seqNo, from)             ⇒ serializeSystemMessageDeliveryAck(seqNo, from)
+    case SystemMessageDelivery.Nack(seqNo, from)            ⇒ serializeSystemMessageDeliveryAck(seqNo, from)
+  }).toByteArray
+
+  override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = manifest match {
+    case QuarantinedManifest ⇒ deserializeQuarantined(ArteryControlFormats.Quarantined.parseFrom(bytes))
+    case ActorSystemTerminatingManifest ⇒ deserializeWithFromAddress(bytes, ActorSystemTerminating)
+    case ActorSystemTerminatingAckManifest ⇒ deserializeWithFromAddress(bytes, ActorSystemTerminatingAck)
+    case HandshakeReqManifest ⇒ deserializeWithFromAddress(bytes, HandshakeReq)
+    case HandshakeRspManifest ⇒ deserializeWithFromAddress(bytes, HandshakeRsp)
+    case ActorRefCompressionAdvertisementManifest ⇒ deserializeCompressionAdvertisement(bytes, deserializeActorRef, ActorRefCompressionAdvertisement)
+    case ActorRefCompressionAdvertisementAckManifest ⇒ deserializeCompressionTableAdvertisementAck(bytes, ActorRefCompressionAdvertisementAck)
+    case ClassManifestCompressionAdvertisementManifest ⇒ deserializeCompressionAdvertisement(bytes, identity, ClassManifestCompressionAdvertisement)
+    case ClassManifestCompressionAdvertisementAckManifest ⇒ deserializeCompressionTableAdvertisementAck(bytes, ClassManifestCompressionAdvertisementAck)
+    case SystemMessageEnvelopeManifest ⇒ deserializeSystemMessageEnvelope(bytes)
+    case SystemMessageDeliveryAckManifest ⇒ deserializeSystemMessageDeliveryAck(bytes, SystemMessageDelivery.Ack)
+    case SystemMessageDeliveryNackManifest ⇒ deserializeSystemMessageDeliveryAck(bytes, SystemMessageDelivery.Nack)
+    case _ ⇒ throw new IllegalArgumentException(s"Manifest '$manifest' not defined for ArteryControlMessageSerializer (serializer id $identifier)")
+  }
+
+  import scala.collection.JavaConverters._
+
+  def serializeQuarantined(quarantined: Quarantined): ArteryControlFormats.Quarantined =
+    ArteryControlFormats.Quarantined.newBuilder()
+      .setFrom(serializeUniqueAddress(quarantined.from))
+      .setTo(serializeUniqueAddress(quarantined.to))
+      .build
+
+  def deserializeQuarantined(quarantined: ArteryControlFormats.Quarantined): Quarantined =
+    Quarantined(deserializeUniqueAddress(quarantined.getFrom), deserializeUniqueAddress(quarantined.getTo))
+
+  def serializeActorRef(ref: ActorRef): String =
+    if (ref == system.deadLetters) "DEADLETTER"
```

Comment:
but I can understand why it's needed if we actually use `deadLetters` ref, since java serialization of `DeadLetterActorRef` does special things